src/platform/Linux/BLEManagerImpl.cpp

```diff
@@ -694,6 +694,8 @@ void BLEManagerImpl::NewConnection(BleLayer * bleLayer, void * appState, const u
     PlatformMgr().ScheduleWork(InitiateScan, static_cast<intptr_t>(BleScanState::kScanForDiscriminator));
 }
 
+void BLEManagerImpl::CancelConnection() {}
```

Comment:
Could we have an implementation or alternatively make CancelConnection return CHIP_ERROR and return NOT_IMPLEMENTED on linux?

linux should be able to cancel the bluez discovery and connect thread. What I implemented a few weeks ago was a scan timeout exactly for this infinite connect reason.

Should darwin also have a timeout until failure?

Reply:
I have updated the code so `CancelConnection` returns `BLE_ERROR_NOT_IMPLEMENTED`. Let me know if you prefer a `CHIP_ERROR_NOT_IMPLEMENTED`.

About darwin, I have added a timeout in https://github.com/project-chip/connectedhomeip/pull/5077 if that's what you were asking for ? 
But this is not enough. What's happening here is that the scanning was still running in the background and the BLE delegate ends up discovering a device and tries to connect to it, but the `RendezvousSession` was freed already.
It may still happens if a device is found before the timeout fires.